### PR TITLE
Add task graph animation to docs homepage

### LINKED ISSUE #10718: Nice visual for docs page
The docs homepage looks like this today:

Screenshot 2023-12-18 at 12 57 57 PM

There's a void above the fold in the upper right of the main contents.  I think it might be useful to have a small visual there, maybe something like the task stream.  @scharlottej13 do we have any good animation content that we can throw up there that you know of?

### PR DESCRIPTION
**Pull request by scharlottej13**
- [x] Closes #10718
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

Screenshot 2023-12-19 at 5 41 31 PM

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -187,6 +187,7 @@ allow_untyped_defs = false
 [tool.codespell]
 ignore-words-list = "coo,nd"
 skip = "docs/source/changelog.rst"
+exclude-file = "docs/source/index.rst"
```
Comment (scharlottej13) [inline pyproject.toml:190]: This is not quite what I was going for... spellcheck doesn't like `medias` on line 21, but when I tried `exclude-file = "docs/source/index.rst:21"` codespell couldn't find the file.

Maybe there's a better way to ignore all .. raw:: html content with a regex or something?

**Comment by mrocklin** [comment]
Fine by me

**Review by jrbourbeau — COMMENTED** [review]
Thanks @scharlottej13 

The spacing looks good on a small screen (though the `...` wrapping in the first bullet point is a little off):

Screenshot 2023-12-20 at 9 51 37 AM

but there's still lots of whitespace on a large screen:

Screenshot 2023-12-20 at 9 51 22 AM

Is it easy to play around with the sizing of the animation to have it display better on a large screen?

**Comment by mrocklin** [comment]
The saved GIF file is 12 MB. I'll probably -1 adding anything that large into the repo.  I'm fine with linking out to Wistia if that works for you.

**Comment by scharlottej13** [comment]
> The saved GIF file is 12 MB. I'll probably -1 adding anything that large into the repo. I'm fine with linking out to Wistia if that works for you.

hmm let me see if the video will work (the mp4 was < 2 MB, but sometimes sphinx is weird about videos)

**Comment by mrocklin** [comment]
Can we just link to an external Wistia thing?

**Comment by mrocklin** [comment]
Thanks @scharlottej13

**Comment by scharlottej13** [comment]
no problem!

**Pull request merged**